test/integration/scheduler/scheduler_test.go

```diff
@@ -611,3 +620,131 @@ func TestNodeEvents(t *testing.T) {
 	}
 
 }
+
+// TestPodSchedulingContextSSA checks that the dynamicresources plugin falls
+// back to SSA successfully when the normal Update call encountered
+// a conflict.
+//
+// This is an integration test because:
+//   - Unit testing does not cover RBAC rules.
+//   - Triggering this particular race is harder in E2E testing
+//     and harder to verify (needs apiserver metrics and there's
+//     no standard API for those).
+func TestPodSchedulingContextSSA(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.DynamicResourceAllocation, true)()
+
+	testCtx := testutils.InitTestAPIServer(t, "podschedulingcontext-ssa", nil)
+	testCtx.DisableEventSink = true
+	testCtx = testutils.InitTestSchedulerWithOptions(t, testCtx, 0)
+	testutils.SyncSchedulerInformerFactory(testCtx)
+	go testCtx.Scheduler.Run(testCtx.SchedulerCtx)
+
+	// NOTE: This test cannot run in parallel, because it is creating and deleting
+	// non-namespaced objects (Nodes).
+	defer testCtx.ClientSet.CoreV1().Nodes().DeleteCollection(testCtx.Ctx, metav1.DeleteOptions{}, metav1.ListOptions{})
```

Comment:
nit: No need to run this -- InitTestAPIServer registers `CleanupTest` via `t.Cleanup()`.
https://github.com/kubernetes/kubernetes/blob/4f2471fc5558dafd97db2426bb1a93f293e617d8/test/integration/util/util.go#L545
https://github.com/kubernetes/kubernetes/blob/4f2471fc5558dafd97db2426bb1a93f293e617d8/test/integration/util/util.go#L344

Reply:
+1
This would also make `linter-hints`  job stop failing with this error:
```
ERROR: test/integration/scheduler/scheduler_test.go:644:59: Error return value of `(k8s.io/client-go/kubernetes/typed/core/v1.NodeInterface).DeleteCollection` is not checked (errcheck)
ERROR: 	defer testCtx.ClientSet.CoreV1().Nodes().DeleteCollection(testCtx.Ctx, metav1.DeleteOptions{}, metav1.ListOptions{})
```

Reply:
I copied this from some other test. Those probably should get cleaned up at some point. For now I'll just remove it here and add error checking for ResourceClasses.

Reply:
@sanposhiho : is it okay now?

Reply:
/lgtm